src/main/kotlin/no/nav/familie/ef/sak/repository/domain/Behandling.kt

```diff
@@ -34,7 +34,10 @@ enum class BehandlingStatus {
     UTREDES,
     FATTER_VEDTAK,
     IVERKSETTER_VEDTAK,
-    FERDIGSTILT,
+    FERDIGSTILT;
+
+    fun behandlingErLåstForVidereRedigering(): Boolean =
```

Comment:
Dette er smart, vi har jo noe lignende hos oss. Er det mulig å legge inn en sjekk for hver save/update på behandling som da kaster feil dersom denne returnerer true tror du? 

Det hadde vært kult om man kunne lagt inn en slik constraint for alt som hadde med behandlingen å gjøre.

Reply:
Det er nok mulig ja, `@PreUpdate` for JPA

For spring-data virker nog dette
```
@Bean
public ApplicationListener<BeforeSaveEvent<Behandling>> loggingSaves() {

	return event -> {

		Behandling behandling = event.getEntity();
                if (behandlingErLåstForVidereRedigering()) error("")
	};
}
```

Reply:
Samtidig brukes den vel fra Vilkår, så kanskje ikke blir veldig smidigt, då må man hente behandlingen først... Går sikkert på noen fin måte :P

Reply:
Kult, dette må jeg sjekke ut!